Mathlib/Data/Int/Card.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Data.Int.Interval
+import Mathlib.Data.Int.ModEq
+import Mathlib.Data.Nat.Count
+import Mathlib.Data.Rat.Floor
+
+/-!
+# Number of elements in an integer/natural number interval congruent to a given number
+
+Pass
+-/
+
+
+open Finset Int
+
+theorem Int.Ico_filter_dvd_eq (a b : ℤ) {r : ℤ} (hr : 0 < r) : (Ico a b).filter (r ∣ ·) =
+    (Ico ⌈a / (r : ℚ)⌉ ⌈b / (r : ℚ)⌉).map ⟨_, mul_left_injective₀ hr.ne'⟩ := by
```

Comment:
Why not
```suggestion
    (Ico (a / r) (b / r)).map ⟨_, mul_left_injective₀ hr.ne'⟩ := by
```
Am I missing an obvious reason why you need to round up and not down?

Reply:
Ah, I guess the open endpoint has to round up, but the closed end doesn't

Reply:
Ah nevermind, I see why

Reply:
Can you add the Ioc version too?

Reply:
Added!